census/src/main/scala/hmda/census/model/Msa.scala

```diff
@@ -0,0 +1,93 @@
+package hmda.census.model
+
+import hmda.model.fi.lar.LoanApplicationRegister
+
+import scala.collection.immutable.ListMap
+import scala.language.implicitConversions
+
+case class Msa(
+    id: String = "",
+    name: String = "",
+    totalLars: Int = 0,
+    totalAmount: Int = 0,
+    conv: Int = 0,
+    FHA: Int = 0,
+    VA: Int = 0,
+    FSA: Int = 0,
+    oneToFourFamily: Int = 0,
+    MFD: Int = 0,
+    multiFamily: Int = 0,
+    homePurchase: Int = 0,
+    homeImprovement: Int = 0,
+    refinance: Int = 0
+) {
+  def addLar(lar: LoanApplicationRegister): Msa = {
+    implicit def bool2int(b: Boolean): Int = if (b) 1 else 0
+
+    new Msa(
+      id = id,
+      name = name,
+      totalLars = totalLars + 1,
+      totalAmount = totalAmount + lar.loan.amount,
+      conv = conv + (lar.loan.loanType == 1),
+      FHA = FHA + (lar.loan.loanType == 2),
+      VA = VA + (lar.loan.loanType == 3),
+      FSA = FSA + (lar.loan.loanType == 4),
+      oneToFourFamily = oneToFourFamily + (lar.loan.propertyType == 1),
+      MFD = MFD + (lar.loan.propertyType == 2),
+      multiFamily = multiFamily + (lar.loan.propertyType == 3),
+      homePurchase = homePurchase + (lar.loan.purpose == 1),
+      homeImprovement = homeImprovement + (lar.loan.purpose == 2),
+      refinance = refinance + (lar.loan.purpose == 3)
+    )
+  }
+}
+
+case class MsaMap(
+    msas: ListMap[String, Msa] = ListMap[String, Msa]()
+) {
+  def +(lar: LoanApplicationRegister): MsaMap = {
```

Comment:
My original concern was related to ordering and pagination.  Before I switched to storing a `Seq` in the ValidationStats, I was worried about relying on the consistency of the iteration order through a `Map`.  `ListMap` guarantees a sorted order based on insertion, but that property is no longer necessary.  I'll make the switch back to an immutable `Map`.